### ISSUE TITLE
factory-auto erroneously accepts mismatching develprj

### ISSUE THREAD

**Issue by jengelh** [openSUSE/scm-staging#52]
Consider https://build.opensuse.org/request/show/1189095 .

The obs source repo is: devel:Factory:git-workflow:staging:ddiss:bcachefs-tools:2
The git source repo is: ddiss/bcachefs-tools

factory-auto should reject it the same way it did to 1189033, because

* the OBS source repo does not match the develprj defined in openSUSE:Factory/bcachefs-tools (which is `filesystems/bcachefs-tools` as of the creation of 1189095)
* the Git source repo does not match the develprj (which would be, after resolving the <scmsync> line, `pool/bcachefs-tools`)

**Comment by jengelh** [comment]
On the other hand, if the develprj "chain of custody" is followed, factory-auto erroneously rejects the submission. Consider this:

```
osc rq list openSUSE:Factory/glslang
1189186  State:review     By:src-o-org-forwarder When:2024-07-23T06:55:09
        Created by: src-o-org-forwarder
        submit:          devel:Factory:git-workflow:staging:jengelh:glslang:1/glslang@1bc7fdc16721be202d9816c2490ee642 -> openSUSE:Factory
        Review by User       is new:       licensedigger                                     
        Review by User       is new:       factory-auto                                      
        Review by Group      is new:       factory-staging                                   
        Descr: glslang 14.3    (🤖: Submission of glslang via
               https://src.opensuse.org/pool/glslang/pulls/1 by jengelh)
```

The Git PR is from `jengelh/glslang` to `pool/glslang:factory` (which maps to `openSUSE:Factory/glslang`).
The develprj of `openSUSE:Factory/glslang` is `X11:Wayland/glslang`, which is equivalent to the state in `jengelh/glslang`.
Therefore, the Git PR is in line with "all submissions must come from the develprj".

factory-auto _erroneously_ rejected the corresponding OBS SR. (Yes, I recognize it came from `devel:Factory:git-workflow:staging:jengelh:glslang:1/glslang`, but its state is also the same as `X11:Wayland/glslang`. And 1189095 was greenlit too.)

It feels like the develprj check in factory-auto is inverted.

**Comment by AdamMajer** [comment]
I agree, the workflow is inverted. I think there were ideas to remove devel projects.

Personally, I would like to keep devel projects as-is. If we have a devel project and the devel project is either git <scmsync> or the package is in <scmsync>, and the head of OBS devel project match the HEAD of the PR, then bot can do a regular `osc sr` from devel project. Upon acceptance, merge to factory branch same sources. Nothing else is needed.

I'll add the logic above for devel projects with <scmsync>.

**Comment by dirkmueller** [comment]
> On the other hand, if the develprj "chain of custody" is followed, factory-auto erroneously rejects the submission. Consider this:

that one has as a comment "Expected a devel project scm sync from https://src.opensuse.org/pool/glslang" which means the devel project entry is not pointing to the pool factory or devel branch. so it is correctly rejected. 

The workflow changed, devel projects are updated *after* factory acceptance, not before.

**Comment by dirkmueller** [comment]
> then bot can do a regular `osc sr` from devel project. Upon acceptance, merge to factory branch same sources. 

No, it can not, because osc sr submits the build service internal representation of the sources, not the sources in git, which are not the same. Which is why the whole workflow change was needed to begin with.

**Comment by AdamMajer** [comment]
> The workflow changed, devel projects are updated _after_ factory acceptance, not before.

Implementing devel projects for Factory requires opposite workflow anyway. But it'll be just very temporary setup.


>>  then bot can do a regular osc sr from devel project. Upon acceptance, merge to factory branch same sources.

> No, it can not, because osc sr submits the build service internal representation of the sources, not the sources in git, which are not the same.

They should be? There is some additional file(s) added, but otherwise the sources are the same.
